va/va.go

```diff
@@ -136,11 +137,48 @@ func (va ValidationAuthorityImpl) getAddr(ctx context.Context, hostname string)
 
 type dialer struct {
 	record core.ValidationRecord
+	stats  metrics.Scope
 }
 
 func (d *dialer) Dial(_, _ string) (net.Conn, error) {
 	realDialer := net.Dialer{Timeout: validationTimeout}
-	return realDialer.Dial("tcp", net.JoinHostPort(d.record.AddressUsed.String(), d.record.Port))
+	conn, err := realDialer.Dial("tcp", net.JoinHostPort(d.record.AddressUsed.String(), d.record.Port))
+
+	// If there was an error connecting to the preferred address and the IPv6First
+	// feature is enabled then attempt another connection if there is a fallback
+	// address available to try
+	if err != nil && features.Enabled(features.IPv6First) {
```

Comment:
Yep. For instance, if there's a connection refused, that means we can reach the IPv6 address, but it's misconfigured. I expect that the most common IPv6 misconfigurations will result in timeouts, and I want to be pretty conservative about what types of misconfigurations we accept.